rs/ethereum/evm-rpc-client/src/types.rs

```diff
@@ -103,6 +103,45 @@ pub mod candid {
         pub topics: Option<Vec<Vec<String>>>,
     }
 
+    #[derive(Clone, Debug, PartialEq, Eq, CandidType, Deserialize)]
+    pub struct FeeHistoryArgs {
+        /// Number of blocks in the requested range.
+        /// Typically, providers request this to be between 1 and 1024.
+        #[serde(rename = "blockCount")]
+        pub block_count: u128,
+
+        /// Highest block of the requested range.
+        /// Integer block number, or "latest" for the last mined block or "pending", "earliest" for not yet mined transactions.
+        #[serde(rename = "newestBlock")]
+        pub newest_block: BlockTag,
+
+        /// A monotonically increasing list of percentile values between 0 and 100.
+        /// For each block in the requested range, the transactions will be sorted in ascending order
+        /// by effective tip per gas and the corresponding effective tip for the percentile
+        /// will be determined, accounting for gas consumed.
```

Comment:
These lines refer to the result rather than the request, right? Maybe this should be made more explicit (as in "The result will contain...")?
Why are the transactions "sorted in ascending order by effective tip per gas"? The order might change quite a bit when computing the actual effective tips, right? Or do I misunderstand this part?

Reply:
This doc is taken 1:1 from the [documentation](https://github.com/ethereum/execution-apis/blob/3ae3d29fc9900e5c48924c238dff7643fdc3680e/src/eth/fee_market.yaml#L40) on `eth_feeHistory`. It aims at explaining (this is how I understand it) how the `reward` field in the response is computed.

Reply:
I see. But I still don't understand it. Is the list sorted by tip per gas or effective tip? Unless the amount of gas is always the same, this isn't the same thing, right?

Reply:
I remember I initially needed to go through the [implementation](https://github.com/ethereum/go-ethereum/blob/766ce23032deae2519322b705e87c7a2ea9f1317/eth/gasprice/feehistory.go#L142) of an EVM-Client to start understanding it, the description [here](https://github.com/ethereum/go-ethereum/blob/766ce23032deae2519322b705e87c7a2ea9f1317/eth/gasprice/feehistory.go#L228) might be also more helpful
```
//   - reward: the requested percentiles of effective priority fees per gas of transactions in each
//     block, sorted in ascending order and weighted by gas used.
```

So back to your question:

1. The transactions are sorted by effective tip per gas
2. The tip is determined by the effective tip for a transaction, such that the sum of all gas consumed by all transactions before and including that one are at most `percentile * gas_used_in_block` (but taking in an additional transaction would overflow that limit)

Does-it make sense?

Reply:
@THLO Since you're on PTO until next Monday and next week I will also be on PTO I asked Manu to approve the MR to be able to merge it. My understanding was that the PR was ok for you but you had some well-founded questions regarding the Rust doc. I hope the above message was able to answer it otherwise feel free to ping me.